scripts/utils/migrate/create-index-pages.js

```diff
@@ -1,96 +1,127 @@
+const shouldSkipDirectory = (dir) => SKIPPED_FOLDERS.includes(dir.path);
 
-    // get files (and links) for this directory
-    const files = nodes.filter((node) => !node.isDirectory());
-    const dirLinks = getPageContent(dir, files);
+const toURL = (node) =>
+  path.join(
+    '/',
+    node.path
+      .replace('.mdx', '')
+      .replace(BASE_DIR, '')
+      .replace(/\/index\/?$/, '')
+  );
 
-    // get all sub directories
-    const subDirs = nodes
-      .filter((node) => node.isDirectory())
-      .map(({ name }) => path.join(dir, name));
-
-    // get links for sub directories (make index pages along the way)
-    const subLinks = await subDirs.reduce(async (contentPromise, subDir) => {
-      // Due to this being async, the previous (accumulator) result is a promise.
-      // We need to await the results of the last loop, before we can add to it.
-      const content = await contentPromise;
-
-      const subDirContent = await createIndexPage(subDir, level + 1);
-      const subDirHeading = getSubheading(subDir, level);
-
-      return [content, subDirHeading, subDirContent].join('\n');
-    }, '');
-
-    // if there is an index.mdx, don't overwrite or make a new one
-    if (files.some((file) => file.name === 'index.mdx')) return;
-
-    // combine links and add them to a new index.mdx file
-    const fileName = `${dir}/index.mdx`;
-    const content = getFrontmatter(title) + dirLinks + subLinks;
-    await fs.writeFile(fileName, content);
+const depthOf = (node, dir) => {
+  return node.path.replace(new RegExp(`${dir.path}\\/`, '')).split('/').length;
+};
 
-    // return the links for this page (to be used by a parent, if applicable)
-    return dirLinks + subLinks;
-  } catch (err) {
-    logger.error(`Unabled to create index for ${dir}: ${err}`);
-  }
+const generateMDX = (dir) => {
+  const tree = root([
+    frontmatter({ title: sentenceCase(dir.basename), template: 'basicDoc' }),
+  ]);
+
+  visit(
+    dir,
+    (node) => node !== dir,
+    (node) => {
+      if (shouldSkipDirectory(node)) {
+        return [visit.SKIP];
+      }
+
+      if (isDirectory(node)) {
+        // Start headings at level 2
+        tree.children.push(
+          heading(depthOf(node, dir) + 1, text(sentenceCase(node.basename)))
+        );
+      } else if (isMDXFile(node)) {
+        const lastChild = last(tree.children);
+        const linkListItem = listItem(
+          link(toURL(node), '', text(node.data.frontmatter.title.trim()))
+        );
+
+        if (isList(lastChild)) {
+          lastChild.children.push(linkListItem);
+        } else {
+          tree.children.push(list('unordered', [linkListItem]));
+        }
+      }
+    }
+  );
+
+  return toMDX(tree);
 };
 
-const createIndexPages = async () => {
-  await createIndexPage(BASE_DIR);
+const createIndexPages = (files) => {
+  const contentDirectory = fromList(
+    files.sort((a, b) => a.path.localeCompare(b.path)),
```

Comment:
what is the `path.localeCompare` doing? Why is this being sorted?

Reply:
Here is the documentation for [localeCompare](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare)

Essentially I wanted the files in alphabetical order so that I could render them as such on each `index.mdx` file.